### PR TITLE
nullifying log functionality when sctp not in create mode

### PR DESCRIPTION
**Pull request by etennoQ**
sets handles of logs to null in sctp when not in create logging mode, this prevents errors when data is passed down handles for files that do not exist

### CODE REVIEW

File: code/segmentedtickerplant/stplog.q [hunk 1]
```diff
@@ -290,14 +290,24 @@ init:{[dbname]
     // add the info to the meta table
     .stpm.updmeta[multilog][`open;logtabs;.z.p+.eodtime.dailyadj];
     ]
+
+  // set log handles to null in sctp if not in create mode
+  if[.sctp.chainedtp and not .sctp.loggingmode=`create; 
+    `..loghandles set .stplg.t!(count .stplg.t)#(::);
+    // only need to log errors if sctp is creating its own logs
+    `.stplg.badmsg set {[x;y;z]};
```
Comment (mpotterq) [inline code/segmentedtickerplant/stplog.q:298]: With a namespace variable you can just use equals, ie.
`.stplg.badmsg:{[x;y;z]};`

File: code/segmentedtickerplant/stplog.q [hunk 2]
```diff
@@ -290,14 +290,24 @@ init:{[dbname]
...
 \d .
 
 // Close logs on clean exit
 .z.exit:{
   if[not x~0i;.lg.e[`stpexit;"Bad exit!"];:()];
-  .lg.o[`stpexit;"Exiting process and closing off log files."];
-  .stpm.updmeta[.stplg.multilog][`close;.stpps.t;.z.p];
-  .stplg.closelog each .stpps.t;
+  .lg.o[`stpexit;"Exiting process"];
+  if[.sctp.loggingmode=`create; // only close logs if sctp is in create mode
```
Comment (mpotterq) [inline code/segmentedtickerplant/stplog.q:308]: This still needs to be triggered for normal STP usage, probably do something like if its an SCTP and mode is not create then `:()`

File: code/segmentedtickerplant/sctp.q [hunk 3]
```diff
@@ -50,6 +50,6 @@ init:{
 
 // Extract data from incoming table as a list
 upd:{[t;x]
-  x:flip value each x;
```
Comment (mpotterq) [inline code/segmentedtickerplant/sctp.q:53]: Is this based on Jonny's comment from earlier?

Comment (etennoQ) [reply code/segmentedtickerplant/sctp.q:53]: yes, sorry unrelated to the logging corrections

Comment (mpotterq) [reply code/segmentedtickerplant/sctp.q:53]: haha no problem!

File: code/segmentedtickerplant/stplog.q (hunk 2 above)
Comment (etennoQ) [reply code/segmentedtickerplant/stplog.q:308]: ah i see what you mean, yeah ill just stop the execution early instead

**Comment by mpotterq** [comment]
Looks good to me!

**Pull request merged**